pkg/observable/channel/map.go

```diff
@@ -15,24 +15,40 @@ type MapFn[S, D any] func(src S) (dst D, skip bool)
 func Map[S, D any](
 	ctx context.Context,
 	srcObservable observable.Observable[S],
-	// TODO_CONSIDERATION: if this were variadic, it could simplify serial transformations.
 	transformFn MapFn[S, D],
 ) observable.Observable[D] {
 	dstObservable, dstProducer := NewObservable[D]()
 	srcObserver := srcObservable.Subscribe(ctx)
 
+	mapInternal[S, D](
+		ctx,
+		srcObserver,
+		transformFn,
+		func(dstNotification D) {
+			dstProducer <- dstNotification
+		},
+	)
+
+	return dstObservable
+}
+
+func mapInternal[S, D any](
```

Comment:
Similar to how you pushed back against `utils` (which I agree with(, I've never been a fan of `Internal`

I'd also argue for:
1. Calling this `goMapNotificationTransformer`
2. Calling `go goMapNotificationTransformer(...)` in the func above
3. Adding a go doc to the lper saying it transforms, optionally skips, and publishes the notification

Reply:
Nice - thanks for calling this out! :sunglasses: